### PR TITLE
Add slippery group for nodes (players/items slide)

### PR DESCRIPTION
**Pull request by Wuzzy2**
Adds the node group “slippery”. Nodes with the group “slippery” set to any value besides 0 will become slippery, players and item entities will slide on it.

This is an attempt to revive #817 by Zeg9. I had to make a few changes to make the code workable. It is based on 569998011e0a9cb6179f1e79fb15b7a3ce80977f which I just pulled in today.

### CODE REVIEW

**Comment by paramat** [comment]
Thanks for taking over on this.

**Comment by Bremaweb** [comment]
It looks like with this slippery is either on or off, there is no way to set how slippery a node is. Would it be difficult to make the slipperiness variable. I would like to be able to make ice really slippery, and snow a little less slippery, and maybe even make gravel a little slippery.

**Comment by o-jasper** [comment]
Oh, i read "value" and assumed a number.. Note that the issue i linked probably would have three parameters... Friction, bounciness and one regarding how it responds to the speed of getting hit. (So that a bouncing ball does not have the same effective friction as a sliding one)

File: builtin/game/item_entity.lua [hunk 1]
```diff
@@ -173,7 +173,7 @@ core.register_entity(":__builtin:item", {
 		local nn = node.name
 		-- If node is not registered or node is walkably solid and resting on nodebox
 		local v = self.object:getvelocity()
-		if not core.registered_nodes[nn] or core.registered_nodes[nn].walkable and v.y == 0 then
+		if not core.registered_nodes[nn] or (core.registered_nodes[nn].walkable and core.get_item_group(nn, "slippery")==0) and v.y == 0 then
```
Comment (SmallJoker) [inline builtin/game/item_entity.lua]: Line too long. Please break it.

File: src/localplayer.cpp [hunk 2]
```diff
@@ -615,7 +615,15 @@ void LocalPlayer::applyControl(float dtime)
 		incH = incV = movement_acceleration_default * BS * dtime;
 
 	// Accelerate to target speed with maximum increment
-	accelerateHorizontal(speedH * physics_override_speed, incH * physics_override_speed);
+	INodeDefManager *nodemgr = m_gamedef->ndef();
+	Map *map = &env->getMap();
+	v3s16 p = floatToInt(getPosition() - v3f(0,BS/2,0), BS);
+	bool slippery = false;
+	try {
+		slippery = itemgroup_get(nodemgr->get(map->getNodeNoEx(p)).groups, "slippery");
```
Comment (SmallJoker) [inline src/localplayer.cpp]: What kind of exception does it throw here?

File: src/localplayer.cpp [hunk 3]
```diff
@@ -627,12 +635,19 @@ v3s16 LocalPlayer::getStandingNodePos()
...
         if (max_increase == 0)
                 return;
 
         v3f d_wanted = target_speed - m_speed;
+	if (slippery)
+	{
+		if (target_speed == v3f(0))
+			d_wanted = -m_speed*.05;
+		else
+			d_wanted = target_speed*.1 - m_speed*.1;
```
Comment (SmallJoker) [inline src/localplayer.cpp]: Code style sould be corrected a bit: `d_wanted = (target_speed - m_speed) * 0.1f;`

**Comment by tobyplowy** [comment]
@Wuzzy2 +1

**Comment by rubenwardy** [comment]
Are you willing to rebase this?

**Comment by Wuzzy2** [comment]
Wait, what? Does this mean the feature just got accepted?

**Comment by jastevenson303** [comment]
Not at all!

**Comment by Fixer-007** [comment]
You need to rebase the code...

**Comment by rubenwardy** [comment]
I'd like to see this in Minetest, but it needs testing thoroughly

**Comment by Wuzzy2** [comment]
Oh crap, I just destroyed my branch! :-(

**Comment by Wuzzy2** [comment]
Whatever, I think this thing needs to be completely re-done anyway because you waited over 18 months before looking at this again …

**Comment by Wuzzy2** [comment]
Take 3:
https://github.com/minetest/minetest/pull/6237

And action!

**Pull request closed without merging**